src/base/bittorrent/torrenthandle.cpp

```diff
@@ -2049,19 +1971,26 @@ void TorrentHandle::adjustActualSavePath()
 
 void TorrentHandle::adjustActualSavePath_impl()
 {
-    QString path;
-    if (!useTempPath()) {
-        // Disabling temp dir
-        // Moving all torrents to their destination folder
-        path = savePath();
-    }
-    else {
-        // Moving all downloading torrents to temporary folder
-        path = m_session->torrentTempPath(info());
-        qDebug() << "Moving torrent to its temporary folder:" << path;
+    const bool needUseTempDir = useTempPath();
+    const QDir tempDir {m_session->torrentTempPath(info())};
+    const QDir currentDir {actualStorageLocation()};
+    const QDir targetDir {needUseTempDir ? tempDir : QDir{savePath()}};
+
+    if (targetDir == currentDir) return;
+
+    if (!needUseTempDir) {
+        if ((currentDir == tempDir) && (currentDir != QDir{m_session->tempPath()})) {
```

Comment:
But QDir performs a smarter comparison to make sure that we don't have two identical directories set by slightly different paths.
Otherwise, we will have to do it manually, but I would not like to worry about it here, so as not to distract from the main task.